### PR TITLE
[RFR] Paginate & sort via query string

### PR DESCRIPTION
**Pull request by manuquentin**

### CODE REVIEW

File: app/Stores/DashboardStore.js [hunk 1]
```diff
@@ -23,6 +24,7 @@ class DashboardStore extends EventEmitter {
         this.data = this.data.update('pending', v => true);
         this.emitChange();
 
+        let {sortDir, sortField} = PathUtils.extractQuery(window.location.hash) || {};
```
Comment (jeromemacias) [inline app/Stores/DashboardStore.js]: Can't we pass values instead of parsing url ?

File: app/Stores/DashboardStore.js [hunk 2]
```diff
@@ -32,11 +34,20 @@ class DashboardStore extends EventEmitter {
             view,
             entity,
             response,
-            entries;
+            entries,
+            dashboardSortField,
+            dashboardSortDir;
```
Comment (jeromemacias) [inline app/Stores/DashboardStore.js]: Can't you simply name it `sort(Field|Dir)` ?

File: app/Stores/ListStore.js [hunk 3]
```diff
@@ -24,11 +26,22 @@ class ListStore extends EventEmitter {
         });
     }
 
-    loadData(configuration, view, page) {
-        page = page || 1;
+    updateParams() {
+        let {sortDir, sortField, page} = PathUtils.extractQuery(window.location.hash) || {};
```
Comment (jeromemacias) [inline app/Stores/ListStore.js]: Same, here, I'm not fan of url parsing.

File: app/Stores/DashboardStore.js (hunk 2 above)
Comment (manuquentin) [inline app/Stores/DashboardStore.js]: Already defined above

File: app/Component/Column/ReferencedList.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import React from 'react';
 import Datagrid from '../Datagrid/Datagrid';
+import PathUtils from 'react-router/lib/PathUtils';
 
 class ReferencedList extends React.Component {
     render() {
         let {entries, field, entityName} = this.props;
+        let {sortDir, sortField} = PathUtils.extractQuery(window.location.hash) || {};
```
Comment (jeromemacias) [inline app/Component/Column/ReferencedList.js]: Is it needed ? Can't you pass them for props ?

File: app/Component/Datagrid/ColumnHeader.js [hunk 5]
```diff
@@ -31,11 +35,16 @@ class ColumnHeader extends React.Component {
 
 ColumnHeader.propTypes = {
     configuration: React.PropTypes.object.isRequired,
+    routeName: React.PropTypes.string.isRequired,
```
Comment (jeromemacias) [inline app/Component/Datagrid/ColumnHeader.js]: It seems not used.

File: app/Component/Datagrid/Datagrid.js [hunk 6]
```diff
@@ -117,13 +116,15 @@ Datagrid.propTypes = {
...
     fields: React.PropTypes.array.isRequired,
     entries: React.PropTypes.array.isRequired,
     sortDir: React.PropTypes.string.isRequired,
-    sortField: React.PropTypes.string.isRequired,
-    view: React.PropTypes.object
+    sortField: React.PropTypes.string.isRequired
+};
+
+Datagrid.contextTypes = {
+    router: React.PropTypes.func.isRequired
```
Comment (jeromemacias) [inline app/Component/Datagrid/Datagrid.js]: Not used.

File: app/View/Dashboard.js [hunk 7]
```diff
@@ -28,8 +28,19 @@ class DashboardView extends React.Component {
         this.setState(DashboardStore.getState());
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.query.page !== this.props.query.page
```
Comment (jeromemacias) [inline app/View/Dashboard.js]: This one will never exists.

File: app/Component/Datagrid/Datagrid.js (hunk 6 above)
Comment (manuquentin) [inline app/Component/Datagrid/Datagrid.js]: Done

**Pull request merged**